File: backend/Services/Parsers/Crosswordle1Parser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class Crosswordle1Parser : BasicTimeResultParser
    {
        private ILogger<Crosswordle1Parser> _logger;

        public Crosswordle1Parser(ILogger<Crosswordle1Parser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        private const string TimePart1 = "time1";
        private const string TimePart2 = "time2";
        public override string GameName => "Crosswordle 1";
        public override string? HelpText => "Reverse engineer a Wordle grid as fast as possible.";
        protected override Regex Parser => new Regex($"Daily Crosswordle \\d+: (?<{TimeGroup}>(?<{TimePart1}>\\d+)[ms] ?(?<{TimePart2}>\\d*)s?)");
        public override string Url => "https://crosswordle.vercel.app/";

        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result.Replace(Url + "?daily=1", string.Empty).Replace("\n\n", "\n").Trim();
        }

        protected override TimeSpan? ParseTimeSpanString(string timeSpan, Match parserResults)
        {
            var timePart1 = parserResults.Groups[TimePart1].Value;
            var timePart2 = parserResults.Groups[TimePart2].Value;

            _logger.LogDebug($"Parsing {timeSpan}...");

            if (timeSpan.Contains("m") && Int32.TryParse(timePart1, out var minutes) && Int32.TryParse(timePart2, out var seconds))
            {
                return new TimeSpan(0, minutes, seconds);
            }
            else if (Int32.TryParse(timePart1, out seconds))
            {
                return new TimeSpan(0, 0, seconds);
            }

            return null;
        }
    }
}

File: lambdas/src/parsers/crosswordle1-parser.ts
import { BasicTimeResultParser } from './base/basic-time-parser';

export class Crosswordle1Parser extends BasicTimeResultParser {
  readonly countWinner = true;
  readonly gameName = 'Crosswordle 1';
  readonly helpText = 'Reverse engineer a Wordle grid as fast as possible.';
  readonly url = 'https://crosswordle.vercel.app/';
  
  // Match "Daily Crosswordle 123: 5m 23s" or "Daily Crosswordle 123: 45s"
  protected readonly parser = new RegExp(
    `Daily Crosswordle \\d+: (?<${BasicTimeResultParser.TIME_GROUP}>(?<timePart1>\\d+)[ms] ?(?<timePart2>\\d*)s?)`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result
      .replace(this.url + '?daily=1', '')
      .replace(/\n\n/g, '\n')
      .trim();
  }
  
  protected parseTimeString(timeStr: string, match: RegExpMatchArray): string | null {
    const timePart1 = match.groups?.timePart1;
    const timePart2 = match.groups?.timePart2;
    
    if (!timePart1) return null;
    
    let minutes = 0;
    let seconds = 0;
    
    // If contains 'm', it's minutes and seconds
    if (timeStr.includes('m') && timePart2) {
      minutes = parseInt(timePart1, 10);
      seconds = parseInt(timePart2, 10);
      if (isNaN(minutes) || isNaN(seconds)) return null;
    } else {
      // Just seconds
      seconds = parseInt(timePart1, 10);
      if (isNaN(seconds)) return null;
    }
    
    return `00:${String(minutes).padStart(2, '0')}:${String(seconds).padStart(2, '0')}`;
  }
}
